driver-core/src/main/com/mongodb/internal/operation/CommandBatchCursor.java

```diff
@@ -93,17 +90,11 @@ class CommandBatchCursor<T> implements AggregateResponseBatchCursor<T> {
         this.firstBatchEmpty = commandCursorResult.getResults().isEmpty();
 
         Connection connectionToPin = null;
-        boolean releaseServerAndResources = false;
-        if (limitReached()) {
-            releaseServerAndResources = true;
-        } else if (connectionDescription.getServerType() == ServerType.LOAD_BALANCER) {
+        if (connectionDescription.getServerType() == ServerType.LOAD_BALANCER) {
```

Comment:
As above: make it a ternary expression at the site of the variable declaration

Reply:
Done